pkg/apis/application/v1alpha1/types.go

```diff
@@ -11,6 +11,48 @@ import (
 	"k8s.io/client-go/rest"
 )
 
+// SyncOperation contains sync operation details.
+type SyncOperation struct {
+	Revision    string               `json:"revision" protobuf:"bytes,1,opt,name=revision"`
+	Prune       bool                 `json:"prune" protobuf:"bytes,2,opt,name=prune"`
+	Overrides   []ComponentParameter `json:"overrides,omitempty" protobuf:"bytes,3,opt,name=overrides"`
+	NoOverrides bool                 `json:"noOverrides,omitempty" protobuf:"bytes,4,opt,name=noOverrides"`
+	DryRun      bool                 `json:"dryRun" protobuf:"bytes,5,opt,name=dryRun"`
+}
+
+// Operation contains requested operation parameters.
+type Operation struct {
+	Sync *SyncOperation `json:"sync" protobuf:"bytes,2,opt,name=sync"`
```

Comment:
As discussed, lets have RollbackOperation as a second field and get rid of NoOverrides. Also why did the protobuf start at 2?

Reply:
Done. Fixed proto number, thank you!